sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -221,6 +221,60 @@ private[sql] object BinaryExpression {
   def unapply(e: BinaryExpression): Option[(Expression, Expression)] = Some((e.left, e.right))
 }
 
+abstract class TernaryExpression extends Expression with trees.TernaryNode[Expression] {
```

Comment:
`Ternary` is a very rare case, and probably most of the code in this class are needed to be overrided for different `TernaryExpression`s, so how about move the logic into the `Conv`?